### PR TITLE
Adds support for batch events

### PR DESCRIPTION
**Pull request by zimbatm**
/cc @mdpye

### CODE REVIEW

File: src/main/java/com/pusher/rest/data/Event.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * POJO for JSON encoding of trigger batch events.
+ */
+public class Event {
+
+    private final String channel;
+    private final String name;
+    private final String data;
+    private final String socket_id;
```
Comment (mdpye) [inline src/main/java/com/pusher/rest/data/Event.java]: Always use camel case. The Gson serialiser is configured to translate to snake case in the serialised output.

**Comment by mdpye** [comment]
To match the existing `trigger` method, which takes an `Object` and serialises it, you should take POJO `Object`s as the data in the `Event` class, and loop over them applying the `serialise` method from `Pusher` to each before giving the list of `Event`s to the `post` method.

File: src/main/java/com/pusher/rest/data/EventBatch.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * POJO body for batch events
+ */
+public class EventBatch {
+
+    private final List<Event> batch;
+
+    public EventBatch(final List<Event> batch) {
+      this.batch = batch;
```
Comment (mdpye) [inline src/main/java/com/pusher/rest/data/EventBatch.java]: Inconsistent indentation

File: src/main/java/com/pusher/rest/data/EventBatch.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    }
```
Comment (mdpye) [inline src/main/java/com/pusher/rest/data/EventBatch.java]: Should have a getter

File: src/test/java/com/pusher/rest/PusherTest.java [hunk 4]
```diff
@@ -96,6 +98,24 @@ protected String serialise(Object data) {
     }
 
     @Test
+    public void batchEvents() throws IOException {
+        context.checking(new Expectations() {{
+            oneOf(httpClient).execute(
+                with(
+                  field(
+                    "batch",
+                    "[{\"channel\": \"my-channel\", \"name\": \"event-name\", \"data\": \"data\"},{\"channel\": \"my-channel\", \"name\": \"event-name\", \"data\": \"data2\", \"socket_id\": \"22.33\"}]"
```
Comment (mdpye) [inline src/test/java/com/pusher/rest/PusherTest.java]: The batch contents aren't double encoded, so you should be matching against a `List` of two `Map`s here, not a string.

File: src/test/java/com/pusher/rest/PusherTest.java [hunk 5]
```diff
@@ -135,6 +155,12 @@ public void channelList() throws Exception {
         p.trigger(channels, "event", Collections.singletonMap("name", "value"));
     }
 
+
+    @Test
+    public void eventList() throws Exception {
+        throw new Exception();
```
Comment (mdpye) [inline src/test/java/com/pusher/rest/PusherTest.java]: This will continue to fail... ;)

File: src/main/java/com/pusher/rest/data/Event.java (hunk 1 above)
Comment (zimbatm) [inline src/main/java/com/pusher/rest/data/Event.java]: Thanks, fixed in efcb2be

File: src/main/java/com/pusher/rest/data/EventBatch.java (hunk 2 above)
Comment (zimbatm) [inline src/main/java/com/pusher/rest/data/EventBatch.java]: Fixed in 67c22d9

File: src/main/java/com/pusher/rest/data/EventBatch.java (hunk 3 above)
Comment (zimbatm) [inline src/main/java/com/pusher/rest/data/EventBatch.java]: Fixed in 2ec68d6

File: src/test/java/com/pusher/rest/PusherTest.java (hunk 4 above)
Comment (zimbatm) [inline src/test/java/com/pusher/rest/PusherTest.java]: Fixed in 4070bfbfb6d1317b9b2958621fccdcd16401f766

File: src/test/java/com/pusher/rest/PusherTest.java (hunk 5 above)
Comment (zimbatm) [inline src/test/java/com/pusher/rest/PusherTest.java]: :) Also fixed in 4070bfbfb6d1317b9b2958621fccdcd16401f766

**Comment by zimbatm** [comment]
Only thing missing is the serialization of the "data" field now

**Pull request merged**